sources/postgres/pgdump.go

```diff
@@ -733,8 +733,11 @@ func getVals(conv *internal.Conv, vll []*pg_query.Node, n *pg_query.InsertStmt)
 		default:
 			conv.Unexpected(fmt.Sprintf("Processing %v statement: found %s in ValuesList", printNodeType(n), printNodeType(vals)))
 		}
+		if len(values) > 0 {
```

Comment:
I'm not sure if the conditional around the append is needed or helpful.

The relevant case is if ​every value in a row hit an error case i.e. one of the default cases in the switches.
​In that case we'd  skip the row.

Now, if we just had a few bad values, then len(values) would be non-zero, and we'd go and append the row, and we'd hit an error when we call ProcessDataRow since we'd find that the number of values and columns didn't match.

So now we have a situation where if there are some errors, we'll handle it in ProcessDataRow, but if all values in a row are errors, then we'll handle it here. That seems confusing. Also it'll break our statistics for rows (ProcessDataRow keeps stats for bad rows and total rows).

Suggest dropping this and unconditionally appending values to rows, and handling all errors in ProcessDataRows.

Reply:
This is a good observation. Thanks for catching this, fixed now.